addons/binding/org.openhab.binding.meterreader/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="meterreader" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:config-description="http://eclipse.org/smarthome/schemas/config-description/v1.0.0"
+    xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+    xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <thing-type id="meter">
+        <label>Meter device</label>
+        <description>The meter device to read the SML or IEC 62056-21 messages from</description>
+
+        <config-description>
+            <parameter name="port" type="text">
+                <label>Connected serial port</label>
+                <description>The device serial port (e.g. /dev/tty0 or COM1)</description>
+                <required>true</required>
+                <limitToOptions>false</limitToOptions>
+                <context>serial-port</context>
+            </parameter>
+            <parameter name="refresh" type="integer">
+                <label>Refresh Rate</label>
+                <description>Refresh rate in seconds</description>
+                <default>20</default>
+                <unitLabel>s</unitLabel>
+            </parameter>
+            <parameter name="baudrateChangeDelay" type="integer">
+                <label>Delay of baudrate change in ms</label>
+                <default>0</default>
+                <description>USB to serial converters often require a delay of up to 250ms after the ACK before changing baudrate</description>
+            </parameter>
+            <parameter name="baudrate" type="text">
+                <label>Baudrate</label>
+                <default>AUTO</default>
+                <description>The baudrate of the serial port. If set to 'AUTO', it is dependent on the selected mode. The default is 300 baud for modes A, B, and C and 2400 baud for mode D, and 9600 baud for SML.</description>
+                <limitToOptions>false</limitToOptions>
+            </parameter>
+            <parameter name="mode" type="text">
+                <label>The protocol mode to use</label>
+                <default>SML</default>
+                <description>Can be SML (PUSH mode), Mode A,B,C (PULL)or D (PUSH)</description>
+            </parameter>
+            <parameter name="conformity" type="text">
+                <label>Conform to specific standard semantics</label>
+                <default>NONE</default>
+                <description>Reserved to conform to special semantics specified in specific standards. EDL_FNN: Currently applies the energy direction to WATT channels (which are absolute values) (see https://www.vde.com/resource/blob/951000/252eb3cdf1c7f6cdea10847be399da0d/fnn-lastenheft-edl-1-0-2010-01-13-data.pdf)</description>
+                <limitToOptions>true</limitToOptions>
```

Comment:
mhh... I didn't provide it here as I have an enum in the code which has the valid values. I'm providing them via a `ConfigOptionProvider`. Otherwise I'd need to maintain both places.

Reply:
Is this ok (what I mentioned), or should I change?

Reply:
Yes, all fine, thanks!